### DISCUSSION TITLE
Why openssl + nghttp3 is still experimental?

### DISCUSSION THREAD

**Discussion by talregev** [curl/curl#13820] [Q&A] [answered]
Why openssl + nghttp3 is still experimental?

**Comment by icing** [comment, marked as answer]
Because performance is poor and uploads often happen at 100% cpu. This is a shortcoming of the OpenSSL API where we have requested changes, but have not seen the necessary improvements so far.

**Comment by talregev** [comment]
Maybe if more people use it, it will bring pressure on openssl team to solve it?

**Reply by bagder** [reply]
I think that's a question for OpenSSL. We have communicated to them what we we think we need from their API in terms of functionality, and it is now up to them what they prioritize.

**Comment by levitte** [comment]
Ping me in the issue in the openssl space, so I can get this on my radar. Please?

**Reply by bagder** [reply]
I believe most of the feedback regarding the current API state has been provided in https://github.com/openssl/openssl/discussions/23339#

**Comment by talregev** [comment]
Now there is a release of openssl 3.5.0.
There is a chance that it will combine with ngtcp2?
